### PR TITLE
Rename Aggregator -> AggregatorHandle; AggregatorFactory -> Aggregator

### PR DESCRIPTION
**Pull request by bogdandrutu**
Left a TODO to move `getInstance` to the new `Aggregator` interface.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregation/AggregationFactory.java [hunk 1]
```diff
@@ -5,11 +5,11 @@
 
 package io.opentelemetry.sdk.metrics.aggregation;
 
-import io.opentelemetry.sdk.metrics.aggregator.Aggregator;
+import io.opentelemetry.sdk.metrics.aggregator.AggregatorHandle;
 import io.opentelemetry.sdk.metrics.common.InstrumentValueType;
 import javax.annotation.concurrent.Immutable;
 
-/** Factory class for {@link Aggregator}. */
+/** Factory class for {@link AggregatorHandle}. */
```
Comment (jkwatson) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregation/AggregationFactory.java:12]: This comment is no longer correct.

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCountAggregator.java [hunk 2]
```diff
@@ -11,77 +11,85 @@
...
+  public MinMaxSumCountAccumulation accumulateDouble(double value) {
+    return MinMaxSumCountAccumulation.create(1, value, value, value);
   }
 
-  private static final class DoubleState {
-    private long count;
-    private double sum;
-    private double min;
-    private double max;
+  static final class Handle extends AggregatorHandle<MinMaxSumCountAccumulation> {
```
Comment (jkwatson) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCountAggregator.java:40]: If we're ok with not testing the exact type returned by the createHandle method, we could make these handle implementations private.

**Review by jkwatson — APPROVED** [review]
Looks good! Thanks!

**Review by anuraaga — APPROVED** [review]

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCountAggregator.java (hunk 2 above)
Comment (bogdandrutu) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCountAggregator.java:40]: I had them initially private, then when I had to fix the tests I did this. We should follow up on this, for the moment I am keeping this way to not remove tests.

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregation/AggregationFactory.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregation/AggregationFactory.java:12]: Updated.

**Pull request merged**